[IMAS-5246] Adhere to udunits and resolve as_parent units (#14)
iterorganization/IMAS-Data-Dictionary: docs/dd_developer_guide.rst

:ref:`dm_rules_guidelines`. In particular sections :ref:`Self-description
Conventions` and :ref:`List of the existing data types`.

NB1: units are normally explicitly defined at the level of the leaf node.

> However, in the case of node belonging to a structure that can be used in different contexts,

Reviewer: ```suggestion
However, in the case of a node belonging to a structure that can be used in different contexts,
```

it's possible to refer to the units of the parent node by indicating ``<units>as_parent</units>``.

> It's also possible to refer to the units of the grand-parent node with 

Reviewer: ```suggestion
It's also possible to refer to the units of the grandparent node with 
```

``<units>as_parent_level_2</units>``. The references will be explicitly resolved 
when generating the ``dd_data_dictionary.xml`` file.

> NB2: ``FLT_*`` and ``CPX_*`` nodes will have sibling errorbar nodes automatically generated

Reviewer: ```suggestion
NB2: ``FLT_*`` and ``CPX_*`` nodes will have sibling errorbar nodes automatically created
```
